### DISCUSSION TITLE
Rewire for SPI

### DISCUSSION THREAD

**Discussion by giovanniborella** [sevmeyer/chrumm-keyboard#4] [Ideas]
This is very promising work.

I am looking into making a pico W - variant (hoping for bluetooth support in curcuitpython down the road) and was wondering is there is a way to free up some pins to allow for SPI-shenannigans.

I notice pin 31-32 are free, so IIC shouldn't be a problem for an OLED. However none of the SPIs are - which would be neat for e-paper

Maybe going

22 -> 31
23 -> 33 (is gnd, but any gnd is fine?)
24 -> 32
25 -> 34

that would free up 21-25 as SPI0, and add a Lipo SHIM to handle battery.

**Comment by sevmeyer** [comment]
I don't have experience with wireless or batteries, so I won't be able to help in that regard.

But yes, logically you can hook up the left-hand columns (22, 24-27, 29) to any other free GPIO pins. Note that according to the RP2040 datasheet, GPIO0 - GPIO22 have enhanced ESD protection, unlike GPIO26 - GPIO28.

The ground pins (23, 28) are not required. They are only included in the column connector because they are placed between the GPIO pins on the Pico. But if one is connected, the other should be connected too.
